src/Http/Ownership/SuggestOwnershipsRequestHandler.php

```diff
@@ -0,0 +1,148 @@
+
+    public function __construct(
+        SearchServiceInterface $searchService,
+        OfferJsonDocumentReadRepository $offerRepository,
+        CurrentUser $currentUser,
+        UserIdentityResolver $userIdentityResolver,
+        DocumentRepository $organizerRepository,
+        OwnershipSearchRepository $ownershipSearchRepository,
+        OrganizerIDParser $organizerIDParser,
+        UserPermissionsReadRepositoryInterface $userPermissionsRepository
+    ) {
+        $this->resultsGenerator = new ResultsGenerator(
+            $searchService,
+            new Sorting('modified', 'desc')
+        );
+        $this->offerRepository = $offerRepository;
+        $this->currentUser = $currentUser;
+        $this->userIdentityResolver = $userIdentityResolver;
+        $this->organizerRepository = $organizerRepository;
+        $this->ownershipSearchRepository = $ownershipSearchRepository;
+        $this->organizerIDParser = $organizerIDParser;
+        $this->userPermissionsRepository = $userPermissionsRepository;
+    }
+
+    public function handle(ServerRequestInterface $request): ResponseInterface
+    {
+        $this->guardAccess();
+
+        $queryParams = new QueryParameters($request);
+
+        $queryParams->guardRequiredEnum('itemType', [ItemType::organizer()->toString()]);
+
+        $id = $this->currentUser->getId();
+        $user = $this->userIdentityResolver->getUserById($id);
+
+        $query = (new SuggestOwnershipsSapiQuery($user))->toString();
+
+        $activeOwnerships = $this->ownershipSearchRepository
+            ->search(new SearchQuery([
+                new SearchParameter('ownerId', $user->getUserId()),
+            ]))
+            ->filter(self::activeOwnerships());
+
+        /**
+         * A map to deduplicate returned organizers
+         * @var array<string, array> $idToOrganizerMap
+         */
+        $idToOrganizerMap = [];
+
+        /**
+         * @var ItemIdentifier $result
+         */
+        foreach ($this->resultsGenerator->search($query) as $result) {
+            $offerType = OfferType::fromCaseInsensitiveValue($result->getItemType()->toString());
+            $id = $result->getId();
+
+            $offer = $this->offerRepository->fetch($offerType, $id)->getAssocBody();
+
+            $organizerId = $this->organizerIDParser->fromUrl(new Url($offer['organizer']['@id']))->toString();
+
+            $ownershipsForOrganizer = $activeOwnerships->filter(fn (OwnershipItem $item) => $item->getItemId() === $organizerId);
+
+            // don't suggest item that already has active ownership
+            if (!$ownershipsForOrganizer->isEmpty()) {
+                continue;
+            }
+
+            // don't suggest duplicate items
+            if (isset($idToOrganizerMap[$organizerId])) {
+                continue;
+            }
+
+            $organizer = $this->organizerRepository->fetch($organizerId)->getAssocBody();
```

Comment:
Why do we get the full organizer JSON LD body? Is it not enough to just get the IDs?

Reply:
The frontend will have to display the title, score, status, image ...

Otherwise the frontend will have to do a fetch for each suggestion, which is also possible

<img width="676" alt="Screenshot 2024-11-18 at 09 35 20" src="https://github.com/user-attachments/assets/f5cd74f9-c3c5-4193-8822-e9abae807d1e">

Reply:
I always have a strong opinion that an API should just provide the building blocks and should not be designed as a convenience for a representation layer. In this case, the UI is the representation layer and in this specific case, it needs information about the organizers. So it's up to the UI to get that extra information.

Reply:
Ok
[Return organizers without assoc body](https://github.com/cultuurnet/udb3-backend/pull/1888/commits/9fda91b11ec413e6c9b828f0469c24c8838ffc93)